publisher/server/publisher.go

```diff
@@ -203,12 +251,21 @@ func max(n []int) int {
 	return val
 }
 
+func (s *Server) debugF(format string, v ...interface{}) {
```

Comment:
I was leaning towards a centralized setup, one logger entry point (a separate package). Unfortunately I am fairly new to the codebase so I don't know off the top of my head what's being used where. If you have a general preference, I can probably get it done.